File: Computer-Seekho-.NET/Computer_Seekho_DN/Service/NewsService.cs
namespace Computer_Seekho_DN.Service;

using System.Collections.Generic;
using System.Threading.Tasks;
using Computer_Seekho_DN.Models;
using Computer_Seekho_DN.Repository;
using Microsoft.AspNetCore.Mvc;
using Microsoft.EntityFrameworkCore;

public class NewsService : INewsService
{
    private readonly ComputerSeekhoDbContext _context;

    public NewsService(ComputerSeekhoDbContext context)
    {
        _context = context;
    }

    public async Task<IEnumerable<News>> GetAllImages()
    {
        return await _context.News.ToListAsync();
    }

    public async Task<News> SaveImage(News image)
    {
        _context.News.Add(image);
        await _context.SaveChangesAsync();
        return image;
    }

    public async Task<News?> GetImageById(int id)
    {
        return await _context.News.FindAsync(id);
    }

    public async Task DeleteImage(int id)
    {
        var news = await _context.News.FindAsync(id);
        if (news != null)
        {
            _context.News.Remove(news);
            await _context.SaveChangesAsync();
        }
    }
}

File: Computer-Seekho-Java/ComputerSeekho/src/main/java/com/Project/Services/NewsService.java
package com.Project.Services;

import java.util.List;
import java.util.Optional;
import com.Project.Entities.News;

import org.springframework.stereotype.Service;

@Service
public interface NewsService {
    public List<News> getAllImages();
    public News saveImage(News image);
    public Optional<News> getImageById(int id);
    public void deleteImage(int id);
    // public List<Image> getbyAlbum(int id);
}
